guava/src/com/google/common/cache/CacheBuilder.java

```diff
@@ -230,13 +230,16 @@ public CacheStats snapshot() {
           });
   static final CacheStats EMPTY_STATS = new CacheStats(0, 0, 0, 0, 0, 0);
 
-  static final Supplier<StatsCounter> CACHE_STATS_COUNTER =
-      new Supplier<StatsCounter>() {
-        @Override
-        public StatsCounter get() {
-          return new SimpleStatsCounter();
-        }
-      };
+  /*
+   * We avoid using a method reference here for now: Inside Google, CacheBuilder is used from the
+   * implementation of a custom ClassLoader that is sometimes used as a system classloader. That's a
+   * problem because method-reference linking tries to look up the system classloader, and it fails
+   * because there isn't one yet.
+   *
+   * I would have guessed that a lambda would produce the same problem, but maybe it's safe because
+   * the lambda implementation is generated as a method in the _same class_ as the usage?
+   */
+  static final Supplier<StatsCounter> CACHE_STATS_COUNTER = () -> new SimpleStatsCounter();
```

Comment:
**Suggestion**: The addition of a detailed comment explaining the reason behind not using method references is helpful for future maintainability. Consider adding references to any relevant issues or discussions that led to this decision for further context. <br> **Code Suggestion**: 
 ```
 +  /*
 +   * Avoiding method references due to potential classloader issues in specific environments. See issue #1234 for a detailed discussion.
 +   */
 ```